src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFCD.cs

```diff
@@ -87,7 +87,12 @@ internal static RegexPrefix Prefix(RegexTree tree)
 
                     case RegexNode.Oneloop:
                     case RegexNode.Onelazy:
-                        if (curNode._m > 0)
+
+                        int cutoff = 1_000_000; // In release, cutoff at a length to which we can reasonably construct a string
```

Comment:
const int Cutoff

Reply:
This would mean people using debug builds and RegexOptions.Debug will need to edit this code and build a private in order to get accurate regex debug tracing.

Reply:
Can you clarify @joshfree ? You mean if they want to get the same cutoff as release, but compiled for debug? I guess so. I think that is rarely going to be needed.

Reply:
Yup.

when investigating a bug in regex you can pass RegexOptions.Debug to a debug build and get trace output.  See https://github.com/dotnet/corefx/blob/master/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs#L132-L155

Reply:
Since we don't ship debug builds, someone doing such an investigation would be building it themselves, at which point they could make this whatever they want if it's a problem. That said, I don't have a preference whether we use a different value for debug or not.

Reply:
Why do we differ between DEBUG and non-DEBUG here? More precisely, why do we disable this optimization for quantifiers > 50 on DEBUG?

Reply:
> Why do we differ between DEBUG and non-DEBUG here?

This kind of thing is done sometimes to increase the chances that both code paths are appropriately tested.  I'm assuming that's what Dan had in mind here.

Reply:
Right. I was once bitten by a case where we called a Win32 API with a buffer that was always large enough in testing, and the "try again with larger buffer" path was bad.